File: channel/php-sdk/models/receiver/entity/order/order_agree_refund_entity.php
<?php

namespace MytChannel\Models\Receiver\Entity\Order;

/**
 * 同意退款请求参数（三方接收麦芽田请求）
 * 麦芽田向三方发送同意退款请求
 * 
 * 使用场景：
 *   麦芽田通知三方同意订单退款
 *
 * 注意事项：
 *   - 此结构用于麦芽田调用三方的同意退款接口
 */
class OrderAgreeRefundReq {
    /**
     * @var string 平台方订单唯一 ID（三方平台的订单号）
     */
    public $order_id;
    
    /**
     * @var string 平台方渠道 ID（三方平台的门店标识）
     */
    public $shop_id;
    
    /**
     * @var string 退款单号
     */
    public $refund_id;
    
    /**
     * @var float 退款金额
     */
    public $refund_amount;
}

/**
 * 同意退款响应数据（三方返回给麦芽田）
 * 三方返回麦芽田的同意退款结果
 */
class OrderAgreeRefundResp {
    /**
     * @var bool 是否处理成功
     */
    public $success;
    
    /**
     * @var string 处理结果描述
     */
    public $message;
    
    /**
     * @var string 平台方订单唯一 ID
     */
    public $order_id;
}

File: channel/go-sdk/models/receiver/entity/order/order_agree_refund_entity.go
// Package entity 提供麦芽田主动推送给三方服务的数据结构定义
package entity

// OrderAgreeRefundReq 同意退款请求参数（麦芽田推送给三方）
// 麦芽田通知三方同意退款
//
// 使用场景：
//
//	麦芽田通知三方同意退款申请
//
// 业务流程：
//
//	麦芽田 → 推送同意退款请求 → 三方处理退款逻辑 → 三方回复处理结果
//
// 注意事项：
//   - 这是麦芽田推送给三方的回调接口
//   - 三方需要实现此接口的回调处理逻辑
type OrderAgreeRefundReq struct {
	OrderId  string `json:"order_id" validate:"required"` // 平台方订单唯一 ID（三方平台的订单号）
	ShopId   string `json:"shop_id" validate:"required"`  // 平台方渠道 ID（三方平台的门店标识）
	RefundId string `json:"refund_id"`                    // 退款单 ID
	Reason   string `json:"reason"`                       // 同意退款原因
}
